### PR TITLE
DBP-351-dbildungs-iam-Helm-chart-for-Keycloak-deployment

### PR DESCRIPTION
**Pull request by sahassou**
# Description
- Utilize Helm templates and helper functions to manage resource names, labels, and selector labels dynamically.
- Include configurations for Deployment, Service, Ingress, and Secret manifests.
- Define a comprehensive _helpers.tpl file to centralize template logic.

## Approval for review

- [ ] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### CODE REVIEW

File: charts/dbildungs-iam/.helmignore [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Patterns to ignore when building packages.
```
Comment (aimee-889) [inline charts/dbildungs-iam/.helmignore:1]: name of the folder should be dbildungscloud-iam-keycloak (same to repo name)

File: charts/dbildungs-iam/dev-realm-spsh.json [hunk 2]
```diff
@@ -0,0 +1,2221 @@
+{
```
Comment (aimee-889) [inline charts/dbildungs-iam/dev-realm-spsh.json:1]: i do not know what this is and if it needs to be here

File: charts/dbildungs-iam/templates/_helpers.tpl [hunk 3]
```diff
@@ -0,0 +1,34 @@
+{{/* Define default values for your chart's name. */}}
+{{- define "dbildungs-iam.name" -}}
```
Comment (aimee-889) [inline charts/dbildungs-iam/templates/_helpers.tpl:2]: please also change dbildungscloud-iam. reference to dbildungscloud-iam-keycloak

File: charts/dbildungscloud-iam-keycloak/Chart.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+apiVersion: v2
+name: dbildungscloud-iam-keycloak
+version: 0.1.1
+
+description: dBildungs-IAM-Keycloak
```
Comment (aimee-889) [inline charts/dbildungscloud-iam-keycloak/Chart.yaml:5]: would write it always in lower case

File: charts/dbildungscloud-iam-keycloak/values.yaml [hunk 5]
```diff
@@ -0,0 +1,4 @@
+replicaCount: 1
+keycloakImage: "ghcr.io/dbildungsplattform/dbildungs-iam:sha-15679d4"  # for test purpose i set this image name
```
Comment (aimee-889) [inline charts/dbildungscloud-iam-keycloak/values.yaml:2]: regarding the default values i am not sure if and how they get set here - i think this depends on the pipeline/structure

**Review by YannickEvers — APPROVED** [review]

**Pull request merged**